aptos-move/block-executor/src/view.rs

```diff
@@ -254,12 +306,39 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceVi
 
                 if matches!(mv_value, ReadResult::Uninitialized) {
                     let from_storage = self.base_view.get_state_value(state_key)?;
+                    let maybe_patched_from_storage = match (from_storage, maybe_layout) {
+                        // There are aggregators / aggregator snapshots in the
+                        // resource, so we have to replace the actual values with
+                        // identifiers.
+                        // TODO(aggregator): gate by the flag.
+                        (Some(state_value), Some(layout)) => {
+                            let res =
+                                self.lift_aggregators_v2_from_state_value(state_value, layout);
+                            if let Err(err) = &res {
+                                // TODO(aggregator): This means replacement failed
+                                //       and most likely there is a bug. Log the error
+                                //       for now, and add recovery mechanism later.
+                                let log_context = AdapterLogSchema::new(
+                                    self.base_view.id(),
+                                    self.txn_idx as usize,
+                                );
+                                alert!(
+                                    log_context,
+                                    "[VM, ResourceView] Error during value exchange for {:?}: {}",
+                                    state_key,
+                                    err
+                                );
+                            }
+                            Some(res?)
+                        },
+                        (from_storage, _) => from_storage,
+                    };
 
                     // This base value can also be used to resolve AggregatorV1 directly from
                     // the versioned data-structure (without more storage calls).
                     state.versioned_map.data().provide_base_value(
                         state_key.clone(),
-                        TransactionWrite::from_state_value(from_storage),
+                        TransactionWrite::from_state_value(maybe_patched_from_storage),
                     );
```

Comment:
Didn't understand this statement. We are creating a transaction write when a resource is read?

Reply:
This is not a write, `provide_base_value` caches the value from storage, and stores it at artificial 0 index (or -1 sort of). But it is not registered as a "write".